File: go/pengyou/model/entity/user_profile.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type UserProfile struct {
	gorm.Model

	UserID        uint      `gorm:"primaryKey" json:"user_id"`
	DisplayName   string    `gorm:"size:50" json:"display_name"`
	AvatarID      string    `gorm:"size:255" json:"avatar_id"`
	Bio           string    `gorm:"size:255" json:"bio"`
	Gender        int8      `json:"gender"`
	Birthday      time.Time `json:"birthday"`
	Location      string    `gorm:"size:100" json:"location"`
	Occupation    string    `gorm:"size:100" json:"occupation"`
	Education     string    `gorm:"size:100" json:"education"`
	School        string    `gorm:"size:100" json:"school"`
	Major         string    `gorm:"size:100" json:"major"`
	Company       string    `gorm:"size:100" json:"company"`
	Position      string    `gorm:"size:100" json:"position"`
	Website       string    `gorm:"size:255" json:"website"`
	CreatedPerson uint      `json:"created_person"`
	UpdatedPerson uint      `json:"updated_person"`
}

func (table *UserProfile) TableName() string {
	return "user_profile"
}

File: java/main/common/src/main/java/com/pengyou/model/entity/UserProfile.java
package com.pengyou.model.entity;

import org.babyfish.jimmer.sql.*;


import org.jetbrains.annotations.Nullable;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * Entity for table "user_profile"
 */
@Entity
public interface UserProfile {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY
    )
    long id();

    @IdView
    long userId();

    @Nullable
    String displayName();

    @Nullable
    String avatarId();

    @Nullable
    String bio();

    @Nullable
    Short gender();

    @Nullable
    LocalDate birthday();

    @Nullable
    String location();

    @Nullable
    String occupation();

    @Nullable
    String education();

    @Nullable
    String school();

    @Nullable
    String major();

    @Nullable
    String company();

    @Nullable
    String position();

    @Nullable
    String website();

    @Nullable
    LocalDateTime createdAt();

    @Nullable
    LocalDateTime modifiedAt();

    @Nullable
    LocalDateTime deleteAt();

    @Nullable
    Long createdPerson();

    @Nullable
    Long modifiedPerson();

    Short modifiedByAdmin();

    @OneToOne
    @JoinColumn(name = "user_id")
    User user();
}

